assets/content/cookbook/Beginner/regular-expressions.md

```diff
@@ -0,0 +1,110 @@
+# Regular expressions
+
+In Haxe a [regular expression](https://en.wikipedia.org/wiki/Regular_expression) starts with `~/` and ends with a single `/` and is of type [`Ereg`](http://api.haxe.org/EReg.html).
+
+```haxe
+var regexp:EReg = ~/world/;
+
+trace(regexp.match("hello world"));
+// true : 'world' was found in the string
+
+trace(regexp.match("hello")); 
+// false : 'world' is not found in the string
+```
+
+Add flags with ending with adding the flag after the ending `/`.
+
+```haxe
+var regexp:EReg = ~/world/i; // case insensitive matching
+
+trace(regexp.match("HELLO WORLD"));
+// true : 'world' was found in the string
+```
+
+String based regular expressions can be writting by creating an actual `EReg` instance:
```

Comment:
It is a little bit weird. I would say:

Regular expressions with a dynamic pattern can be created by using the `EReg` constructor as follows:
```haxe
var world = "world";
var regexp:EReg = new EReg("hello " + world, "i"); // case insensitive matching 

trace(regexp.match("HELLO WORLD"));
// true : 'hello world' was found in the string
```